R/WIP_parameters_selection.R

```diff
@@ -0,0 +1,64 @@
+#' Parameters Selection
+#'
+#' Automated selection of the 'best' parameters.
+#'
+#' @param model A statistical model
+#' @param method The method used in the variable selection. Can be NULL (default), "forward" or "L1". See \code{projpred::varsel}.
+#' @param cross_validation Select with cross-validation.
+#'
+#' @examples
+#' \dontrun{
+#' library(rstanarm)
+#' model <- stan_glm(mpg ~ ., data = mtcars, chains = 2, iter = 200)
+#' parameters_selection(model)
+#'
+#' model <- stan_glm(mpg ~ cyl * disp * hp * drat, data = mtcars, chains = 2, iter = 200)
+#' parameters_selection(model, cross_validation = TRUE)
+#' }
+#'
+#'
+#'
+#' @export
+parameters_selection <- function(model, method=NULL, cross_validation = FALSE, ...){
+  if (!requireNamespace("projpred", quietly = TRUE)) {
+    stop("Package `projpred` required. Please install it by running `install.packages(projpred)`.", call. = FALSE)
+  }
+
+  if(cross_validation){
+    message("Cross-validating best parameters...")
+    junk <- capture.output(selection <- projpred::cv_varsel(model, method=method))
+  } else{
+    selection <- projpred::varsel(model, method=method)
+  }
+
+  # Visualise
+  # varsel_plot(selection, stats = c('elpd', 'rmse'), deltas=T)
+
+  # Extract parameters
+  projection <- projpred::project(selection, nv = projpred::suggest_size(selection))
+  parameters <- names(as.data.frame(as.matrix(projection)))
+
+  formula <- .reconstruct_formula(parameters)
+  formula
+}
+
+
+
+
+
+
+#' @keywords internal
+.reconstruct_formula <- function(parameters){
```

Comment:
@strengejacke I wonder if there is a way of directly getting the "formula" instead of the parameters. If not, I wonder an exported function `construct_formula` or something (in insight?) could be useful to reconstruct an original and compact formula based on the parameters...

Reply:
Do you mean find_formula()?

Reply:
No really, if I'm not correct find_formula retrieve a formula from models and/or from terms, whereas here we would need to reconstruct the formula from the list of parameters. For instance from this:

```r
[1] "(Intercept)" "cyl"         "disp"        "cyl:disp"    "hp"          "sigma"    
```

to `cyl * disp + hp` 

Which gets tricky with triple or partial interactions